tests/test_3pl_logistic_company/tests/test_computed_fields.py

```diff
@@ -0,0 +1,73 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields
+from odoo.tests.common import TransactionCase
+from odoo.tests import tagged
+
+
+@tagged('post_install', '-at_install')
+class ComputedFieldsTestCase(TransactionCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.partner_1 = cls.env['res.partner'].create({
+            'name': 'Test Partner 1',
+        })
+        cls.partner_2 = cls.env['res.partner'].create({
+            'name': 'Test Partner 2',
+        })
+        cls.product = cls.env['product.template'].create({
+            'name': 'Test Product',
+            'type': 'consu',
+        })
+        cls.stock_picking = cls.env['stock.picking'].create({
+            'location_id': cls.env.ref('stock.stock_location_stock').id,
+            'location_dest_id': cls.env.ref('stock.stock_location_customers').id,
+            'picking_type_id': cls.env.ref('stock.picking_type_in').id,
+        })
+        cls.move_line = cls.env['stock.move.line'].create({
+            'picking_id': cls.stock_picking.id,
+            'product_id': cls.product.id,
+            'quantity': 1,
+        })
+        cls.quality_check = cls.env['quality.check'].create({
+            'picking_id': cls.stock_picking.id,
+            'move_line_id': cls.move_line.id,
+        })
+
+    def test_x_quality_check_owner_sale_order_line_computation_when_move_line_has_owner(self):
+        self.move_line.write({
+            'owner_id': self.partner_1.id,
+        })
+        self.assertEqual(self.quality_check.x_quality_check_owner, self.partner_1, "Quality check owner should be the same as the owner of the picking")
```

Comment:
I wonder if this is correct. The compute method would only be triggered if a field in depends (move_line_id, picking_id) changes, but here it does not. So I'm pretty sure it only works due to some caching issue. If you compute it before changing the partner on the move line, I bet this line would fail.
Maybe the trigger should be adapted and be (move_line_id.owner_id, picking_id.owner_id)